### ISSUE TITLE
9.1.3.1a/3.1 Although the page has some headings, these only partially help to structure the page content as the page has two "h1" headings, and directly following the second "h1" is an "h5" heading. Ideally, a page should have only one "h1" heading; multiple "h1" headings are only allowed when there are multiple main topics on the page.

### ISSUE THREAD

**Issue by szaimen** [nextcloud/server#42269]
9.1.3.1a/3.1 Although the page has some headings, these only partially help to structure the page content as the page has two "h1" headings, and directly following the second "h1" is an "h5" heading. Ideally, a page should have only one "h1" heading; multiple "h1" headings are only allowed when there are multiple main topics on the page.

![image](https://github.com/nextcloud/server/assets/42591237/5bd2faa2-7fea-44f0-9af0-e3518b52ed9c)

**Comment by JuliaKirschenheuter** [comment]
@ShGKme could this one be related https://github.com/nextcloud/server/issues/37092?

**Comment by ShGKme** [comment]
> @ShGKme could this one be related #37092?

No. Any number of `H*` elements in any order is valid in an HTML document.

![image](https://github.com/nextcloud/server/assets/25978914/f493aec9-2c9a-422a-a55a-fc7d7c422359)

**Comment by emoral435** [comment]
@JuliaKirschenheuter should we close this issue, then ?

**Comment by JuliaKirschenheuter** [comment]
> @JuliaKirschenheuter should we close this issue, then ?

No, the issue is still there. It is a valid HTML, yes, but it is still not ok that we have `h1` -> `h3`, `h1` -> `h5`, `h2` -> `h5`, and so on. We have to make sure that every page we have have a clear heading structure `h1` -> `h2`, `h2` -> `h3`, `h3` -> `h4`. More about https://www.w3.org/WAI/tutorials/page-structure/headings/

**Comment by JuliaKirschenheuter** [comment]
Dear `@michaelnissenbaum ` // replace after 06.01

We are working on this issue through different pages. I have a question regarding this one page with a Text app. There are 2 kind of headings: 1. "All files -> (name of the file)", 2. Content of the text file with all possible variations: `h1 -> h6`, `checkboxes` ... The content depends only on the user input, so we can't prevent several `<h1>` or illogical heading sequence. 

The question is: should we remove the first part with file name if the file is on editing mode? I guess this info is important for the user to understand where he actually located.

![Screenshot from 2024-01-02 10-21-28](https://github.com/nextcloud/server/assets/6078378/d5931e7b-5a77-4465-8919-272f54c35921)

Thank you!

**Comment by JuliaKirschenheuter** [comment]
https://github.com/nextcloud/server/issues/42269#issuecomment-1873779675 -> will be handled here https://github.com/nextcloud/server/issues/42275#issuecomment-1889117511

**Comment by szaimen** [comment]
I think this one is rather
